packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -507,56 +494,72 @@ export class ProjectService {
     }
 
     async getJobStatus(jobUuid: string): Promise<Job> {
-        return this.jobModel.getJobstatus(jobUuid);
-    }
-
-    async startJob(projectUuid: string | undefined): Promise<string> {
-        const jobUuid = uuidv4();
-
-        await this.jobModel.upsertJobStatus(
-            jobUuid,
-            projectUuid,
-            JobStatusType.STARTED,
-        );
-
-        return jobUuid;
+        return this.jobModel.get(jobUuid);
     }
 
-    async getAllExplores(
+    async compileProject(
         user: SessionUser,
         projectUuid: string,
-        jobUuid?: string,
-        forceRefresh: boolean = false,
-    ): Promise<(Explore | ExploreError)[]> {
-        const cachedExplores = await this.projectModel.getExploresFromCache(
+    ): Promise<{ jobUuid: string }> {
+        const job: CreateJob = {
+            jobUuid: uuidv4(),
+            jobType: JobType.COMPILE_PROJECT,
+            jobStatus: JobStatusType.STARTED,
             projectUuid,
-        );
-        if (!cachedExplores || cachedExplores.length === 0 || forceRefresh) {
-            await this.projectModel.tryWithProjectLock(
-                projectUuid,
-                jobUuid || uuidv4(),
-                this.jobModel,
-                async () => {
-                    const explores = await this.refreshAllTables(
-                        user,
-                        projectUuid,
+            steps: [{ stepType: JobStepType.COMPILING }],
+        };
+        return new Promise<{ jobUuid: string }>((resolve, reject) => {
+            const onLockFailed = async () => {
+                reject(
+                    new AlreadyProcessingError('Project is already compiling'),
+                );
+            };
+            const onLockAcquired = async () => {
+                await this.jobModel.create(job);
+                resolve({ jobUuid: job.jobUuid });
+                try {
+                    await this.jobModel.update(job.jobUuid, {
+                        jobStatus: JobStatusType.RUNNING,
+                    });
+                    const explores = await this.jobModel.tryJobStep(
+                        job.jobUuid,
+                        JobStepType.COMPILING,
+                        async () => this.refreshAllTables(user, projectUuid),
                     );
                     await this.projectModel.saveExploresToCache(
                         projectUuid,
                         explores,
                     );
-                },
-            );
-        }
-        return cachedExplores || [];
+                    await this.jobModel.update(job.jobUuid, {
+                        jobStatus: JobStatusType.DONE,
+                    });
+                } catch (e) {
+                    await this.jobModel.update(job.jobUuid, {
```

Comment:
Do we need to call `await this.jobModel.setPendingJobsToSkipped(job.jobUuid);` as we do on create.